### PR TITLE
Updated code for Signal Generators and PulseBlaster.

### PR DESCRIPTION
**Pull request by ichristen**
This pull request:

- Includes the changes in #138 (`pulseblaster`), as Experiments and Sources in `signalgen` that use PulseBlasters were added and changed, and it makes sense to add both branches at the same time.
- Implements modernization of Sources to use Prefs.
- Adds old (yet modernized) Signal Generator code.
- Adds old (and updated yet not modernized) microwave Experiment code.

### CODE REVIEW

File: +Prefs/+Inputs/BooleanField.m [hunk 1]
```diff
@@ -3,14 +3,32 @@
 
     properties(Hidden)
         uistyle = 'checkbox';
+        nanuistyle = 'radiobutton';
```
Comment (ichristen) [inline +Prefs/+Inputs/BooleanField.m:6]: Changes to Prefs.Boolean to allow_nan. NaN is displayed as a circular (and readonly) radiobutton to distinguish it from the square checkbox on or checkbox off.

**Comment by ebersin** [comment]
@ichristen can this be broken up into smaller commits? This is a lot of code.

**Comment by ichristen** [comment]
> @ichristen can this be broken up into smaller commits? This is a lot of code.

Somewhat, @ebersin .

- Experiment code (~1000 lines) can be added separately (original plan was to add them separately, in fact).
- However, everything else is kinda interconnected. Keep in mind that the bulk of the reminding is signal generator driver code and associated UI functions to connect to them (which I have not edited, these were already on the branch).

**Comment by ichristen** [comment]
Closing request due to splitting into several smaller requests.

**Pull request closed without merging**